libs/init/migrate/sbt/exportscript/src/mill/init/migrate/sbt/ExportSbtBuildScript.scala

```diff
@@ -0,0 +1,225 @@
+            case r =>
+              println(s"skipping repository $r")
+              None
+          }
+        )
+      ) ++ Seq(
+        if ((Keys.publish / Keys.skip).value) None
+        else Some(PublishModuleConfig(
+          pomSettings = PublishModuleConfig.PomSettings(
+            description = Keys.description.value,
+            organization = Keys.organization.value,
+            url = Keys.homepage.value.fold("")(_.toExternalForm),
+            licenses = Keys.licenses.value.map {
+              case (name, url) => PublishModuleConfig.License(name, name, url.toExternalForm)
+            },
+            versionControl = Keys.scmInfo.value.fold(PublishModuleConfig.VersionControl()) {
+              scmInfo =>
+                import scmInfo._
+                PublishModuleConfig.VersionControl(
+                  Some(browseUrl.toExternalForm),
+                  Some(connection),
+                  devConnection
+                )
+            },
+            developers = Keys.developers.value.map { developer =>
+              import developer._
+              PublishModuleConfig.Developer(id, name, url.toExternalForm)
+            }
+          ),
+          publishVersion = Keys.version.value,
+          versionScheme = Keys.versionScheme.value
+        ))
+      ).flatten
+      val testConfigs =
+        if (
+          (Test / Keys.unmanagedSourceDirectories).value.exists(_.exists()) ||
+          (Test / Keys.unmanagedResourceDirectories).value.exists(_.exists())
+        )
+          Seq(JavaModuleConfig(
+            mvnDeps = mvnDeps(_.contains("test")),
+            moduleDeps = moduleDeps(cs => Seq("test", "test->compile").exists(cs.contains)) ++
+              moduleDeps(_.contains("test->test")).map(dep =>
+                dep.copy(dep.segments :+ testModuleName)
+              ),
+            compileModuleDeps =
+              moduleDeps(cs => Seq("test->provided", "test->optional").exists(cs.contains)),
+            runModuleDeps = moduleDeps(_.contains("test->runtime"))
+          ))
+        else Nil
+
+      SbtModule(
+        moduleDir = moduleDir.subRelativeTo(os.pwd).segments,
+        baseDir = baseDir.subRelativeTo(os.pwd).segments,
+        platformCrossType = crossPlatformBaseDir.map { dir =>
+          if (os.exists(dir / "shared")) "Full"
+          else if (os.exists(dir / "src")) "Pure"
+          else "Dummy"
+        },
+        crossScalaVersions = Keys.crossScalaVersions.value,
+        useVersionRanges =
+          isCrossScalaVersion && os.walk.stream(moduleDir).exists(path =>
+            os.isDir(path) && path.last.matches("""^scala-\d+\.\d*(-.*|\+)$""")
+          ),
+        mainConfigs = mainConfigs,
+        testModuleName = testModuleName,
+        testModuleBase = mvnDependencies.iterator.map(
+          _.organization
+        ).collectFirst(ModuleTypedef.testSupertypeByDepOrg),
+        testConfigs = testConfigs
+      )
+    }
+  }
```

Comment:
Where does the JSON serialization for these case classes happen? I assume they must be serialized to extract them out of SBT, but I can't seem to find where the code for that happens

Reply:
See `exportModules` defined above this method.